target/linux/layerscape/files/arch/arm64/boot/dts/freescale/traverse-ls1043s.dts

```diff
@@ -0,0 +1,339 @@
+ *     Software is furnished to do so, subject to the following
+ *     conditions:
+ *
+ *     The above copyright notice and this permission notice shall be
+ *     included in all copies or substantial portions of the Software.
+ *
+ *     THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ *     EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ *     OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ *     NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ *     HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ *     WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ *     FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ *     OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/dts-v1/;
+#include "fsl-ls1043a.dtsi"
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	model = "Traverse LS1043S";
+	compatible = "traverse,ls1043s";
+
+	aliases {
+		crypto = &crypto;
+		ethernet0 = &EMAC0;
+		ethernet1 = &EMAC1;
+		ethernet2 = &EMAC2;
+		ethernet3 = &EMAC3;
+		ethernet4 = &EMAC4;
+		ethernet5 = &EMAC5;
+	};
+	
+	leds {
+		compatible = "gpio-leds";
+		gpio0 {
+			label = "ls1043s:green:user0";
+			gpios = <&gpio1 23 GPIO_ACTIVE_HIGH>;
+		};
+		gpio1 {
+			label = "ls1043s:green:user1";
+			gpios = <&gpio1 24 GPIO_ACTIVE_HIGH>;
+		};
+		/* LED D17 */
+		gpio2 {
+			label = "ls1043s:green:wan";
+			gpios = <&gpio1 26 GPIO_ACTIVE_LOW>;
+		};
+		gpio3 {
+			label = "ls1043s:yellow:wan";
+			gpios = <&gpio1 27 GPIO_ACTIVE_LOW>;
+		};
+		/* LED D18 */
+		gpio4 {
+			label = "ls1043s:green:mgmt";
+			gpios = <&gpio1 28 GPIO_ACTIVE_LOW>;
+		};
+		gpio5 {
+			label = "ls1043s:yellow:mgmt";
+			gpios = <&gpio1 29 GPIO_ACTIVE_LOW>;
+		};
+		/* LED D6 */
+		gpio6 {
+			label = "ls1043s:green:user2";
+			gpios = <&gpio1 31 GPIO_ACTIVE_HIGH>;
+		}; 
+
+		teng_act {
+			label = "ls1043s:yellow:10gact";
+			gpios = <&gpio4 0 GPIO_ACTIVE_HIGH>;
+		};
+
+		teng_link {
+			label = "ls1043s:green:10glink";
+			gpios = <&gpio4 1 GPIO_ACTIVE_HIGH>;
+		};
+	};
+	
+	gpio-keys-polled {
+		compatible = "gpio-keys-polled";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		poll-interval = <1000>;
+		/* This button may not be loaded on all boards */
+		button@0 {
+			label = "Front button";
+			linux,code = <KEY_SETUP>;
+			gpios = <&gpio1 25 GPIO_ACTIVE_LOW>;
+		};
+		/* This is wired to header S3 */
+		button@1 {
+			label = "Rear button";
+			linux,code = <KEY_WPS_BUTTON>;
+			gpios = <&gpio1 30 GPIO_ACTIVE_LOW>;
+		};
+	}; 
+};
+/* Only for Rev 1 boards and later
+&esdhc {
+	mmc-hs200-1_8v;
+	sd-uhs-sdr104;
+	sd-uhs-sdr50;
+	sd-uhs-sdr25;
+	sd-uhs-sdr12;
+};
+*/
+
+&i2c0 {
+	status = "okay";
+	rtc@6f {
+		compatible = "intersil,isl1208";
+		reg = <0x6f>;
+	};
+
+	sfp_pca9534: pca9534@24 {
+		compatible = "ti,tca9534", "nxp,pca9534";
+		gpio-controller;
+		#gpio-cells = <2>;
+		reg = <0x24>;
+		gpio-base = <100>;
+		/* 
```

Comment:
I was hoping we could use this to 'name' GPIOs rather than /etc/config/system, unfortunately GPIO-hog means they can't be re-exported by userspace. I'll get rid of them.